component/component.go

```diff
@@ -37,7 +37,8 @@ type Component interface {
 	// Create a new context from the context.Background() for long-running operations.
 	Start(ctx context.Context, host Host) error
 
-	// Shutdown is invoked during service shutdown.
+	// Shutdown is invoked during service shutdown. After Shutdown() is called, if the component accept data in
```

Comment:
`Shutdown signals the component to stop accepting data and to clean up the resources in use, as the collector is being stopped.`

Reply:
If we want to clarify this we also need to tell that any internally stored pdata must be drained, propagated further to the pipeline to the next consumer (or exported out of the Collector if the component is exporter) before Shutdown returns. 
"Clean up the resources" hints to that but I think it is not sufficient.

Reply:
@jpkrohling I would remove the "as the collector is being stopped" that sounds to me too much information from upstream. How about saying "as the processor is being decommissioned"?

@tigrannajaryan this is outlined in https://github.com/open-telemetry/opentelemetry-collector/pull/2504/files#diff-6f065ffd54f98081ffd548c200264f761067d883b72b315c01ac28ef5a8fce47R45-R46 but I like the "before Shutdown returns" part. I think I should add it.